### PR TITLE
`FlightIcon` => `Hds::Icon` Internal migration - `showcase` & `website`

### PR DESCRIPTION
**Pull request by zamoore**
### :pushpin: Summary

If merged, this PR updates instances of `FlightIcon` within `website` & `showcase` to use the `Hds::Icon` component.

### :link: External links

Jira ticket: [HDS-3707](https://hashicorp.atlassian.net/browse/HDS-3707)
Jira ticket: [HDS-3708](https://hashicorp.atlassian.net/browse/HDS-3708)

### Known Percy issues
- Tooltip - Text change
Screenshot 2024-08-20 at 9 35 46 AM
- Icon - Added a border to the icons in the "Display" section that was missing previously
Screenshot 2024-08-20 at 9 36 43 AM

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3707]: https://hashicorp.atlassian.net/browse/HDS-3707?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HDS-3708]: https://hashicorp.atlassian.net/browse/HDS-3708?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by aklkv** [comment]
A few spots you might want to remove references to ember-flight-icons packages:

https://github.com/hashicorp/design-system/blob/64a0cc615e5b2e55506f762b573e5c6e2b5a82f1/website/package.json#L20

https://github.com/hashicorp/design-system/blob/64a0cc615e5b2e55506f762b573e5c6e2b5a82f1/website/package.json#L51

**Comment by Dhaulagiri** [comment]
Unless there is an objection from someone, I would target these straight to `main` as there shouldn't be a need to stage them in a feature branch.

File: .changeset/soft-badgers-obey.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (Dhaulagiri) [inline .changeset/soft-badgers-obey.md:1]: issue (blocking) we don't version the showcase or the website, so this file can be deleted

**Comment by Dhaulagiri** [comment]
Within the showcase, it looks like there are several instances where we have styles targeting `.flight-icon` that should be updated. I think that will resolve most of the regressions seen in Percy.

**Comment by zamoore** [comment]
> Within the showcase, it looks like there are several instances where we have styles targeting `.flight-icon` that should be updated. I think that will resolve most of the regressions seen in Percy.

@didoo @alex-ju - The remaining ones are necessary. This is ready for re-review

**Comment by Dhaulagiri** [comment]
> The remaining ones are necessary

Can you expand a bit on why so that we understand a bit better?

**Comment by zamoore** [comment]
> > The remaining ones are necessary
> 
> Can you expand a bit on why so that we understand a bit better?

Sure, the last 2 examples of it being targeted with CSS are in `showcase/app/styles/showcase-pages/icon.scss` and `showcase/app/styles.scss`. These two are still used to style the `FlightIcon` examples we have at foundations/icon

**Comment by Dhaulagiri** [comment]
Shouldn't those be getting converted as well, or are we leaving them behind as examples for some reason?

**Comment by alex-ju** [comment]
> Shouldn't those be getting converted as well, or are we leaving them behind as examples for some reason?

Agree with Brian. things like `showcase/app/templates/foundations/icon.hbs` should be removed at this point as we don't plan any further changes to the `ember-flight-icons`

**Comment by zamoore** [comment]
> > Shouldn't those be getting converted as well, or are we leaving them behind as examples for some reason?
> 
> Agree with Brian. things like `showcase/app/templates/foundations/icon.hbs` should be removed at this point as we don't plan any further changes to the `ember-flight-icons`

Don't we still want the showcase page so we can run tests until FlightIcon is removed?

**Comment by Dhaulagiri** [comment]
> things like showcase/app/templates/foundations/icon.hbs should be removed at this point as we don't plan any further changes to the ember-flight-icons

Recognizing that I've sent us into a bit of a loop on this, I would think we'd want this page to still exist but use `<Hds::Icon>` moving forward?

**Review by Dhaulagiri — APPROVED** [review]
👏

**Review by alex-ju — APPROVED** [review]

**Pull request merged**